src/service/backup_service.rs

```diff
@@ -0,0 +1,135 @@
+use std::sync::Arc;
+
+use crate::{
+    persistence::{backup::BackupStoreApi, db::SurrealDbConfig, identity::IdentityStoreApi},
+    util,
+};
+
+use super::{Error, Result};
+#[cfg(test)]
+use mockall::automock;
+
+/// Allows to backup and restore the database as an encrypted file.
+#[cfg_attr(test, automock)]
+#[async_trait::async_trait]
+pub trait BackupServiceApi: Send + Sync {
+    /// Creates an encrypted backup of the database and returns the
+    /// download path to the backup file.
```

Comment:
same in the comment - doesn't return a path, but rather the bytes

Reply:
Yes that was a rework :grinning: will update